ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/rest/OkHttpRestClientTest.java

```diff
@@ -189,6 +195,21 @@ void postsAsync() throws InterruptedException {
     assertThat(request.getMethod()).isEqualTo("POST");
   }
 
+  @Test
+  void postsAsyncDoesNotThrowExceptionsInOtherThreadsWhenRequestCreationFails() {
+
+    final TestObject2 requestBodyObject = new TestObject2(TEST_BLOCK_HASH);
+    final SafeFuture<Response<TestObject>> responseFuture =
+        underTest.postAsync(
+            TEST_PATH, requestBodyObject, failingRequestTypeDefinition, responseTypeDefinition);
+
+    // this will fail if exceptions are thrown in other threads
+    await().catchUncaughtExceptions().atMost(Duration.ofSeconds(1)).until(responseFuture::isDone);
```

Comment:
Awaitility can be really finicky to get right so we generally use it via the `Waiter` class so that if we need to fix some source of intermittency we can fix it in one class and it's solved everywhere.  So I'd suggest moving this call into a new method in Waiter (and then I think you won't need the direct dependency on awaitility).

I'd also say that the maximum wait of 1 second is too tight - the system can be very busy when running tests so we usually want a long timeout (like 30 seconds) - it should still finish as soon as responseFuture is done which keeps it running quickly when things are working.

Reply:
Thanks, did a change to `Waiter`, good to know of that class for the future :)